### DISCUSSION TITLE
Power plant capability

### DISCUSSION THREAD

**Discussion by Tthermo** [BioSTEAMDevelopmentGroup/biosteam#198] [Q&A] [answered]
Process engineering is not my jam but I've started to dabble so forgive me if this is an obvious question. Is there an intrinsic or particular reason biosteam is leveled specifically at biorefineries? With a few more units for things like Gibbs free energy minimization or a bioreactor to combustion reactor hack, a very large additional power generation research space is unlocked. When combustion is considered, like in the BoilerTurboGenerator, its in a relatively fixed fuel LHV/HHV and fixed unit arrangement capacity.

I'm specifically interested in starting with some simple rankine cycle exploration and everything in biosteam seems like a dream except for the combustion bit. I plan on doing some hacking to get what I need but I don't expect it to be at the code level of the current biosteam package. That being said, I can upload my efforts if anyone thinks it will at all be useful, even if someone just wants to see what interests at least one random guy.

Thanks for all your work.

**Comment by yalinli2** [comment, marked as answer]
Thanks for the interest! BioSTEAM was first developed for biorefineries because the funding that supported the developers are from the [CABBI](https://cabbi.bio/) (Center for Advanced Bioenergy and Bioprodcuts Innovation), so we've been mainly working on bio-related projects :)

But certainly feel free to use it for things that you need and we'd love to see your contributions, thanks in advance!

Tagging @yoelcortes and see if he has things to add on

**Comment by yoelcortes** [comment]
@Tthermo, thank you for your kind post and interest! @yalinli2, thanks for helping out so much with Q&As! 

It is in our mission to expand biosteam applications and it is certainly possible for thermal systems/cycles. In fact, enthalpy, entropy, and Gibbs-free-energy are already chemical and stream properties.

External developments to BioSTEAM in the past have been driven by specific needs of other groups. I try to make supporting outside contributions a priority and the process should not take long. Once you form a pull request, I would work closely with you to meet standards on documentation/testing/speed.

Thanks,

**Reply by Tthermo** [reply]
Thanks for the response. I'm excited to give it a whirl when I get a spot of free time. After I have things running well enough Ill do a formal pull request.

**Reply by Tthermo** [reply]
Also, I see you are at Madison. If you ever get a chance to work with Prof Kokjohn, he is a cool guy. (He was my advisor.)

**Reply by farhan26mh** [reply]
Hi, are you working on gasification/ Gibbs free energy minimization? I am struggling in creating a unit for Gibbs free energy and I am that one random guy who would be super grateful if you kindly shared your work so that I would get some ideas! Having some sleepless nights because of this so would mean a lot!

**Reply by Tthermo** [reply]
No way. I was working on some other stuff first but I can put some more attention into this if I'm not the only one interested. Don't expect anything quickly, but I will try to have something I can share in the next couple weeks.

**Comment by yoelcortes** [comment]
This is so cool, looking forward to help with the implementation.
